File: LeetCode/0206-reverse-linked-list/0206-reverse-linked-list.java
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode reversed = null;
        
        while (head != null) {
            ListNode next = head.next;
            head.next = reversed;
            reversed = head;
            head = next;
        }
        
        return reversed;
    }
}

File: LeetCode/0206-reverse-linked-list/0206-reverse-linked-list.kt
class Solution {
    fun reverseList(head: ListNode?): ListNode? {
        return reverse(head, null)
    }
    
    fun reverse(head: ListNode?, prev: ListNode?): ListNode? {
        if (head == null) return prev
        
        val next = head.next
        
        head.next = prev
        
        return reverse(next, head)
    }
}